git_machete/git_operations.py

```diff
@@ -125,7 +125,7 @@ def of(value: str) -> Optional["FullCommitHash"]:
 
     @staticmethod
     def is_valid(value: str) -> bool:
-        return value is not None and len(value) == 40
+        return value is not None and len(value) == 40 and value.isalnum()
```

Comment:
Still not enough! What about `g` repeated 40 times? still not a valid SHA1 hash as they can only contain hex digits